src/Components/Components/src/RenderHandle.cs

```diff
@@ -49,6 +49,11 @@ public Dispatcher Dispatcher
         /// </summary>
         public bool IsHotReloading => HotReloadFeature.IsSupported && (_renderer?.IsHotReloading ?? false);
 
+        /// <summary>
+        /// Gets whether the renderer has been disposed.
+        /// </summary>
+        internal bool IsRendererDisposed => _renderer?.Disposed ?? false;
+
```

Comment:
There are times need to re-render `<SectionOutlet>` content when a corresponding `<SectionContent>` gets disposed. However, there are also times when the renderer itself gets disposed before these components are (e.g. transitioning out of prerendering, refreshing the page). I'm open to undoing this change if we find another way to identify when the renderer can no longer be used.